sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -632,6 +632,7 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
   object BasicOperators extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
+      case AlreadyPlanned(physical, output) => AlreadyPlannedExec(physical, output) :: Nil
```

Comment:
Maybe I am not following completely, but what happens if you return the physical plan here? Or do you get into trouble with rules that are not exactly idempotent?

Reply:
You still go through the entire join planning, Adaptive query planning and much more. We also may need to add projections to rewrite expression ids in case the plan was re-used in joins, etc

Reply:
I don't think that we want to rewrite expression IDs. This should only be used for cases where we need to pass a physical plan as a logical plan within the same physical plan's execution. That's the case for `InsertableRelation`. We want all of those physical nodes to remain unchanged so that metrics are correctly reported to the UI.

Reply:
So we should make sure the "already planned" `SparkPlan` comes from `sparkPlan` not `executedPlan`, right? @hvanhovell 
And indeed some physical preparation rules are not idempotent, WSCG being one of them.